packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -105,6 +112,33 @@ function AllTransactions({ account = {}, transactions, filtered, children }) {
     }));
   }, [schedules, accountId]);
 
+  let runningBalance = useMemo(() => {
+    if (!showBalances) {
+      return 0;
+    }
+
+    // balances is in descending order so latest order is at the start of the array.
+    return balances ? Object.values(balances)[0]?.balance ?? 0 : 0;
+  }, [showBalances, balances]);
+
+  let prependBalances = useMemo(() => {
+    if (!showBalances) {
+      return null;
+    }
+
+    // Reverse so we can calculate from earliest upcoming schedule.
+    let scheduledBalances = prependTransactions.map((_, index, ptArray) => {
+      let scheduledTransaction = ptArray[ptArray.length - 1 - index];
+      let amount =
+        (scheduledTransaction._inverse ? -1 : 1) * scheduledTransaction.amount;
+      return {
+        balance: (runningBalance += amount),
+        id: scheduledTransaction.id,
+      };
+    });
```

Comment:
Couldn’t this use `reverse()` to be a bit more normal reading?

```suggestion
    let scheduledBalances = prependTransactions.map((scheduledTransaction) => {
      let amount =
        (scheduledTransaction._inverse ? -1 : 1) * scheduledTransaction.amount;
      return {
        balance: (runningBalance += amount),
        id: scheduledTransaction.id,
      };
    }).reverse();
```

Reply:
Updated. I thought I'd avoid the extra allocation.